### ISSUE TITLE
1.0 RC: Build failure on s390x

### ISSUE THREAD

**Issue by olebole** [gnudatalanguage/gdl#676]
When building the package on the s390x platform (IBM zSystems), I get the following error while compiling `sorting.cpp`:
```
cd "/<<PKGBUILDDIR>>/obj-s390x-linux-gnu/src" && /usr/bin/c++  -DHAVE_CONFIG_H -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -Dgnudatalanguage_EXPORTS -I/usr/include/GraphicsMagick -I/usr/lib/s390x-linux-gnu/wx/include/gtk3-unicode-3.0 -I/usr/include/wx-3.0 -I/usr/include/geotiff -I/usr/include/hdf5/serial -I/usr/include/hdf -I/usr/include/python3.7m -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/eigen3 -I"/<<PKGBUILDDIR>>" -I"/<<PKGBUILDDIR>>/obj-s390x-linux-gnu"  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -DBUILD_DATE="\"Dec  4 2019\"" -std=gnu++11 -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC   -fopenmp -std=gnu++11 -o CMakeFiles/gnudatalanguage.dir/sorting.cpp.o -c "/<<PKGBUILDDIR>>/src/sorting.cpp"
/<<PKGBUILDDIR>>/src/sorting.cpp: In function ‘T* lib::RadixSort(const DInt*, SizeT)’:
/<<PKGBUILDDIR>>/src/sorting.cpp:105:21: error: ‘H1_OFFSET2’ was not declared in this scope; did you mean ‘H1_OFFSET8’?
  105 |  T* h1 = &Histogram[H1_OFFSET2]; /* Histogram for last pass (MSB) */                  \
      |                     ^~~~~~~~~~
/<<PKGBUILDDIR>>/src/sorting.cpp:913:5: note: in expansion of macro ‘CREATE_HISTOGRAMS2’
  913 |     CREATE_HISTOGRAMS2(DInt, input);
      |     ^~~~~~~~~~~~~~~~~~
/<<PKGBUILDDIR>>/src/sorting.cpp: In function ‘T* lib::RadixSort(const DUInt*, SizeT)’:
/<<PKGBUILDDIR>>/src/sorting.cpp:105:21: error: ‘H1_OFFSET2’ was not declared in this scope; did you mean ‘H1_OFFSET8’?
  105 |  T* h1 = &Histogram[H1_OFFSET2]; /* Histogram for last pass (MSB) */                  \
      |                     ^~~~~~~~~~
/<<PKGBUILDDIR>>/src/sorting.cpp:989:5: note: in expansion of macro ‘CREATE_HISTOGRAMS2’
  989 |     CREATE_HISTOGRAMS2(DUInt, input);
      |     ^~~~~~~~~~~~~~~~~~
make[3]: *** [src/CMakeFiles/gnudatalanguage.dir/build.make:1873: src/CMakeFiles/gnudatalanguage.dir/sorting.cpp.o] Error 1
```
The reason is that `H1_OFFSET2` is not defined on big endian systems, https://github.com/gnudatalanguage/gdl/blob/296da47ed4f2b9d91f567ef45befe37495012857/src/sorting.cpp#L58-L60

but only on little endian systems, https://github.com/gnudatalanguage/gdl/blob/296da47ed4f2b9d91f567ef45befe37495012857/src/sorting.cpp#L75-L78

Could the missing definition be added here?

**Comment by maynardGK** [comment]
The line defining H1_OFFSET2 got lost when IS_BIGENDIAN was inserted.
(#654 ) corrects this and updates the windows' version, also.
````
#if defined(IS_BIGENDIAN)
    #define H1_OFFSET2  0
  ...
````

**Comment by olebole** [comment]
Adding this resolves the build failure on s390x (not really a surprise).

**Comment by GillesDuvert** [comment]
patched in #684 
Thanks!
